src/py4dgeo/py4dgeo_python.cpp

```diff
@@ -126,12 +126,18 @@ PYBIND11_MODULE(_py4dgeo, m)
     "nearest_neighbors",
     [](const KDTree& self, EigenPointCloudConstRef cloud) {
       KDTree::NearestNeighborsDistanceResult result;
-      self.nearest_neighbors_with_distances(cloud, result);
+      int k = 1;
+      self.nearest_neighbors_with_distances(cloud, result, k);
 
-      return std::make_tuple(as_pyarray(std::move(result.first)),
-                             as_pyarray(std::move(result.second)));
+      py::list result_list;
+      for (size_t i = 0; i < result.size(); ++i) {
+        result_list.append(std::make_tuple(std::move(result[i].first[0]),
+                                           std::move(result[i].second[0])));
+      }
```

Comment:
This is a potentially large data structure (as large as the point cloud). Therefore, it should not be returned as a Python list (see Liams performance comparison: https://github.com/ssciwr/pybind11-numpy-example/) but as a numpy array. There is two ways of fixing: Either construct and return a numpy array here in the pybind code or expose the `k` parameter to the Python side and fix its usage in ICP.